src/zarr/core/array.py

```diff
@@ -3449,3 +3456,305 @@ def _get_default_codecs(
+        Whether to overwrite an array with the same name in the store, if one exists.
+    config : ArrayConfig or ArrayConfigParams, optional
+        Runtime configuration for the array.
+    data : np.ndarray, optional
+        Initial data for the array.
+
+    Returns
+    -------
+    z : array
+        The array.
+    """
+
+    if zarr_format is None:
+        zarr_format = _default_zarr_version()
+
+    # TODO: figure out why putting these imports at top-level causes circular imports
+    from zarr.codecs.sharding import ShardingCodec
+
+    # TODO: fix this when modes make sense. It should be `w` for overwriting, `w-` otherwise
+    mode: Literal["a"] = "a"
+    dtype_parsed = parse_dtype(dtype, zarr_format=zarr_format)
+    config_parsed = parse_array_config(config)
+    shape_parsed = parse_shapelike(shape)
+    chunk_key_encoding_parsed = _parse_chunk_key_encoding(
+        chunk_key_encoding, zarr_format=zarr_format
+    )
+    store_path = await make_store_path(store, path=name, mode=mode, storage_options=storage_options)
+    shard_shape_parsed, chunk_shape_parsed = _auto_partition(
+        shape_parsed, shard_shape, chunk_shape, dtype_parsed
+    )
+    result: AsyncArray[ArrayV3Metadata] | AsyncArray[ArrayV2Metadata]
+
+    if zarr_format == 2:
+        if shard_shape_parsed is not None:
+            msg = (
+                'Zarr v2 arrays can only be created with `shard_shape` set to `None` or `"auto"`.'
+                f"Got `shard_shape={shard_shape}` instead."
+            )
+
+            raise ValueError(msg)
+        if filters != "auto" and not all(isinstance(f, numcodecs.abc.Codec) for f in filters):
+            raise TypeError(
+                "For Zarr v2 arrays, all elements of `filters` must be numcodecs codecs."
+            )
+        filters = cast(Iterable[numcodecs.abc.Codec] | Literal["auto"], filters)
+        filters_parsed, compressor_parsed = _parse_chunk_encoding_v2(
+            compression=compression, filters=filters, dtype=dtype_parsed
+        )
+        if dimension_names is not None:
+            raise ValueError("Zarr v2 arrays do not support dimension names.")
+        if order is None:
+            order_parsed = zarr_config.get("array.order")
+        else:
+            order_parsed = order
+
+        result = await AsyncArray._create_v2(
+            store_path=store_path,
+            shape=shape_parsed,
+            dtype=dtype_parsed,
+            chunks=chunk_shape_parsed,
+            dimension_separator=chunk_key_encoding_parsed.separator,
+            fill_value=fill_value,
+            order=order_parsed,
+            filters=filters_parsed,
+            compressor=compressor_parsed,
+            attributes=attributes,
+            overwrite=overwrite,
+            config=config_parsed,
+        )
+    else:
+        array_array, array_bytes, bytes_bytes = _get_default_encoding_v3(dtype_parsed)
+        sub_codecs = (*array_array, array_bytes, *bytes_bytes)
```

Comment:
this seems to ignore any compression/filter that is passed

Reply:
oops, that's definitely not intended!

Reply:
this is now fixed